contributing/development/cpp_usage_guidelines.rst

```diff
@@ -53,18 +53,61 @@ as Godot provides its own data types (among other things).
++------------------------+--------------------------+--------------------------------------------------------------------------------------+
+| ``Array`` 📜           | ``std::vector``          | Values can be of any Variant type. No static typing is imposed.                      |
+|                        |                          | Uses shared reference counting, similar to ``std::shared_ptr``.                      |
++------------------------+--------------------------+--------------------------------------------------------------------------------------+
+| ``TypedArray`` 📜      | ``std::vector``          | Subclass of ``Array`` but with static typing for its elements.                       |
+|                        |                          | Not to be confused with ``Packed*Array``, which is internally a ``Vector``.          |
++------------------------+--------------------------+--------------------------------------------------------------------------------------+
+| ``Packed*Array`` 📜    | ``std::vector``          | Alias of ``Vector``, e.g. ``PackedColorArray = Vector<Color>``.                      |
+|                        |                          | Only a limited list of packed array types are available                              |
+|                        |                          | (use ``TypedArray`` otherwise).                                                      |
++------------------------+--------------------------+--------------------------------------------------------------------------------------+
+| ``List``               | ``std::list``            | Linked list type. Generally slower than other array/vector types. Prefer using       |
+|                        |                          | other types in new code, unless using ``List`` avoids the need for type conversions. |
++------------------------+--------------------------+--------------------------------------------------------------------------------------+
+| ``Span``               | ``std::span``            | Represents read-only access to a contiguous array without needing to copy any data.  |
+|                        |                          | See `pull request description <https://github.com/godotengine/godot/pull/100293>`__  |
+|                        |                          | for details.                                                                         |
++------------------------+--------------------------+--------------------------------------------------------------------------------------+
+| ``HashSet``            | ``std::unordered_set``   | **Use this as the "default" set type.**                                              |
++------------------------+--------------------------+--------------------------------------------------------------------------------------+
+| ``RBSet``              | ``std::unordered_set``   | Uses a `red-black tree <https://en.wikipedia.org/wiki/Red-black_tree>`__             |
```

Comment:
```suggestion
| ``RBSet``              | ``std::set``             | Uses a `red-black tree <https://en.wikipedia.org/wiki/Red-black_tree>`__             |
```
This was correctly in that sense, also I think a description of `RBMap` is needed as well, might also want to describe `VSet` and `VMap`, which are equivalent to `std::flat_set/map` (which are in C++23, or in Boost)

Reply:
`VSet` and `VMap` also notably use copy-on-write semantics. But they aren't used much in the engine at all, and the benefit is unclear, so it has been proposed to remove them again. 
I guess there's no harm in covering them since they do exist, but I'd probably attach a note that the benefits aren't clearly established.